### DISCUSSION TITLE
Jekyll-build-pages running despite .nojekll file

### DISCUSSION THREAD

**Discussion by travisdowns** [community/community#23564] [Other Feature Feedback, Questions, & Ideas]
<p>Recently I’ve seen new actions appearing in my githup pages repo, which runs Jekyll on my github pages site. <a href="https://github.com/actions/jekyll-build-pages" rel="noopener nofollow ugc">This</a> is the action involved, and <a href="https://github.blog/changelog/2021-12-16-github-pages-using-github-actions-for-builds-and-deployments-for-public-repositories/" rel="noopener nofollow ugc">this</a> is presumably the announcement of this “feature”.</p>
<p>I already have my own actions CI setup to run a (much newer) version of Jekyll, with plugins beyond the whitelisted set, and so on. I have no idea what the effect of running a second, differently configured and different version of Jekyll on the output files will be.</p>
<p>I do not want to run Jekyll a second time on the output files of my CI. How I can disable this? I have a <code>.nojekyll</code> file in the root of the repo, which used to disable jekyll, but it is ignored.</p>

**Comment by YiMysty** [comment, marked as answer]
<p>:wave: if you are publishing pages from <code>docs</code> folder, the <code>.nojekyll</code> file should be put under <code>docs</code> folder instead of repo’s root folder</p>

**Comment by yoannchaudet** [comment]
<p>Hello <a class="mention" href="/travisdowns">@travisdowns</a>,</p>
<p>Like <a class="mention" href="/yimysty">@YiMysty</a> pointed below, your <code>.nojekyll</code> file is in the wrong folder.</p>
<p>Even before we moved to running your build/deployment on Actions that means Jekyll was running on your static assets.</p>
<p>If you relocate the <code>.nojekyll</code> file to your <code>/docs</code> folder on the branch that is building, you will notice that the build step is skipped and instead only the archive step runs as expected.</p>
<p>Let us know if you run into further issues!</p>

**Comment by travisdowns** [comment]
<p>Thanks, that was exactly it!</p>
<p>It did cross my mind that this might be the case, but everything I could find said that the file should be in the root of the repository (but perhaps this is just because publishing out of the docs folder is uncommon). It doesn’t seem like there is any good documentation on <code>.nojekyll</code> from GitHub itself, just third-party sources, so maybe a couple of sentences in the doc would help.</p>
<p>After making the change, I no longer get the pointless Jekyll build.</p>
<p>I do look forward to the day I can customize the behavior of the bot, because the way it is now the flow is fairly redundant for my use case: I do the whole build and upload the artifact then push it to <code>gh-pages</code> branch, then the bot syncs it out, builds again an artifact, and pushes it to GH pages. If I could just push it directly or have the bot do it I’d save about 30 seconds in the push-to-live timing and also GitHub would save resources not storing two copies of the site as artifacts, etc.</p>

**Comment by yoannchaudet** [comment]
<p><a class="mention" href="/travisdowns">@travisdowns</a> Good news, your hopes will come true in a few months. We are actively working on exactly that!</p>

**Reply by VitamintK** [reply]
did this end up happening?

**Comment by travisdowns** [comment]
<p>Thanks Yoann, looking forward to it!</p>
<p>Just wanted to add, in case anyone else runs across this thread: I <em>did</em> actually have a <code>.nojekyll</code> file in the root of my blog source directory, so in principle you might expect it to have copied that file to the output (as it does for most files without a <code>_</code> prefix). However, files starting with <code>.</code> are excluded by default, so to fix this I had to add an <code>include: [".nojekyll"]</code> in my <code>_config.yml</code>, like so:</p>
<aside class="onebox githubcommit" data-onebox-src="https://github.com/travisdowns/travisdowns.github.io/commit/f710ae2972309bd29f08d02747da9ebb2e9573dd">
  <header class="source">

      <a href="https://github.com/travisdowns/travisdowns.github.io/commit/f710ae2972309bd29f08d02747da9ebb2e9573dd" target="_blank" rel="noopener nofollow ugc">github.com/travisdowns/travisdowns.github.io</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/travisdowns/travisdowns.github.io/commit/f710ae2972309bd29f08d02747da9ebb2e9573dd" target="_blank" rel="noopener nofollow ugc">include .nojekyll file in _site</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2022-01-24" data-time="03:03:19" data-timezone="UTC">03:03AM - 24 Jan 22 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/travisdowns" target="_blank" rel="noopener nofollow ugc">
          <img alt="travisdowns" src="https://user-images.githubusercontent.com/2403521/181077924-900e3801-02f0-46c9-9ce9-29bdb7321abf.jpeg" class="onebox-avatar-inline" width="20" height="20">
          travisdowns
        </a>
      </div>

      <div class="lines" title="changed 1 files with 3 additions and 0 deletions">
        <a href="https://github.com/travisdowns/travisdowns.github.io/commit/f710ae2972309bd29f08d02747da9ebb2e9573dd" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+3</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">Including this file will present GitHub from redundantly running
Jekyll on the g<span class="show-more-container"><a href="https://github.com/travisdowns/travisdowns.github.io/commit/f710ae2972309bd29f08d02747da9ebb2e9573dd" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">enerated site (we have our own build pipeline).</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by dmnsgn** [comment]
Wasn't it working previously if the `.nojekyll` was in a subfolder?
